vllm/v1/core/single_type_kv_cache_manager.py

```diff
@@ -289,15 +277,34 @@ def remove_skipped_blocks(self, request_id: str,
         # No need to remove blocks for full attention.
         pass
 
-    def get_num_common_prefix_blocks(self, request_id: str,
-                                     num_running_requests: int) -> int:
-        blocks = self.req_to_blocks[request_id]
+    def get_num_common_prefix_blocks(
+            self, running_request_id: str, running_request_ids: list[str],
+            transfering_request_ids: list[str]) -> int:
+        """Get common prefix blocks shared by all running and transferring 
+        requests."""
+        if running_request_id not in self.req_to_blocks:
+            return 0
+
+        all_request_ids = running_request_ids + transfering_request_ids
+        request_blocks = [
+            self.req_to_blocks[req_id] for req_id in all_request_ids
+            if req_id in self.req_to_blocks
+        ]
+
+        reference_blocks = self.req_to_blocks[running_request_id]
+        total_requests = len(all_request_ids)
+
         num_common_blocks = 0
-        for block in blocks:
-            if block.ref_cnt == num_running_requests:
+        for i, ref_block in enumerate(reference_blocks):
+            requests_with_block = sum(
+                1 for blocks in request_blocks if i < len(blocks)
```

Comment:
please only iterate over transferring requests. Iterate over running requests is slow.

Reply:
If we don't iterate over running requests, then I think it's a little hard to ensure the complete logical correctness of this flow. I may be wrong here, but in my opinion, the below should work in most cases and I have done the same in the new changes that I pushed.
- If ref_block.ref_cnt >= total_requests (at least total_requests reference it)
- And all transferring requests have it
- Then, all running requests must also have it

Please correct me if there is a flaw in my logic or understanding of the flow. I will update the code changes accordingly. Thanks, and please take a look at this @heheda12345